File: cpp/cpp_100-999/274.cc
// C++
// leetcode 274
// https://leetcode-cn.com/problems/h-index/
// 简简单单排个序，有大到小的，然后就出来了

class Solution {
public:
	auto static cmp(int a, int b) {  return a > b;}

    int hIndex(vector<int>& citations) {
    	sort(citations.begin(), citations.end(), cmp);
    	int n = citations.size();

    	int h = 0;
    	for(h = 0; h < n; ++h) {
    		if(citations[h] < h + 1)
    			break;
    	}

    	return	h;
    }
};

File: Go/Go_100-999/274.go
// Go
// leetcode 274
// https://leetcode.cn/problems/h-index/

func hIndex(citations []int) int {
	cnt := make([]int, 1005)
	for _, x := range citations {
		cnt[x]++
	}

	sum := 0
	for i:=1000; i>=0; i-- {
		sum += cnt[i]
		if sum >= i {
			return i
		}
	}
	return 0
}